test/bundle_tag_parser_test.dart

```diff
@@ -0,0 +1,37 @@
+@TestOn('browser')
+
+import 'dart:typed_data';
+
+import 'package:arweave/src/utils/bundle_tag_parser.dart';
+import 'package:arweave/src/models/tag.dart';
+import 'package:arweave/utils.dart';
+import 'package:test/test.dart';
+
+import 'snapshots/data_bundle_test_snaphot.dart';
+import 'deserialize_tags.dart';
+
+void main() {
+  group('bundle tag parser', () {
+    test('check if avro serializes tags correctly', () {
+      final buffer = serializeTags(tags: testTagsSnapshot);
+      expect(buffer, equals(testTagsBufferSnapshot));
+    });
+
+    test('check if avro fails serialization when wrong data is given', () {
```

Comment:
```suggestion
    test('throws an exception when bad input data is provided', () {
```
It's not clear to me what's goin on in this test... Are the tags actually bad tags? Is the serialized buffer not correct? Is the way we're passing that buffer to the deserialize function incorrect? Also, should we be analyzing the exception more deeply to ensure that the one we expect is actually thrown?

Reply:
From your suggested test name, we can understand that what we're testing here is not if the tags are actually bad, but the buffer that we use as input for `deserializeTags` is in the wrong format. Knowing that, I updated the test name and also improved the test now expecting the specific error type that `deserializeTags` output in this case.

[e2d2f3d](https://github.com/ardriveapp/arweave-dart/pull/31/commits/e2d2f3d307fb4a09259ab3906ac327fadceaa426)

Reply:
I think this could be made even clearer with an update to the test name... You stated:
> but the buffer that we use as input for deserializeTags is in the wrong format
How exactly is it in the wrong format?